quodlibet/quodlibet/ext/events/libraryfileorganizing.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Jonathan Schmeling
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import os
+import re
+import shutil
+from gi.repository import Gtk
+from quodlibet import _
+from quodlibet import app
+from quodlibet import util
+from quodlibet import qltk
+from quodlibet.qltk.entry import UndoEntry
+from quodlibet.pattern import Pattern
+from quodlibet.plugins import PluginConfigMixin
+from quodlibet.plugins.events import EventPlugin
+from quodlibet.util.library import get_scan_dirs
+from quodlibet.util.library import scan_library
+
+
+class MyPlugin(EventPlugin, PluginConfigMixin):
+    PLUGIN_ID = "libraryfileorganizing"
+    PLUGIN_NAME = _("Automatically Update Library Files")
+    PLUGIN_DESC = _("After making any update (namely, to tags), update "
+                    "the file to a new location/name based on its tags in "
+                    "a form that you specify. This is most useful for "
+                    "automatically keeping organized a central music "
+                    "directory.")
+
+    DEFAULT_PAT = "~/Music/<albumartist|<albumartist>|" + \
+                  "<artist|<artist>|Unknown Artist>>/" + \
+                  "<album|<album>|Unknown Album>/" + \
+                  "<albumartist|<albumartist>|" + \
+                  "<artist|<artist>|Unknown Artist>> " + \
+                  "(<album|<album>|Unknown Album>) - " + \
+                  "<~#track|<~#track><~#disc| (<~#disc>)> - >" + \
+                  "<title|<title>|Unknown Title>>"
+    CFG_PAT_PLAYING = "naming_pattern"
+
+    def plugin_on_changed(self, songs):
+        song = songs[0]
+        current_location = song("~filename")
+        file_split = os.path.splitext(current_location)
+        file_extension = file_split[1]
+        psbl_new_location = os.path.abspath(
+            os.path.expanduser((Pattern(self.config_get(self.CFG_PAT_PLAYING,
+                                                        self.DEFAULT_PAT)) %
+                                song) + file_extension))
+
+        if(not(psbl_new_location == re.sub(" \\([0-9]+\\)$",
+                                           "",
+                                           file_split[0]) + file_extension)):
+            dirname = os.path.dirname(psbl_new_location)
+            if(not(os.path.exists(dirname))):
```

Comment:
To make our codestyle somewhat consistent, please add a space after keywords like "if" and "not" and only use outer parenthesis on multi-line statements (the linter should have warned about this...). So `if not os.path.exists(dirname):` in this case.